### PR TITLE
feat: update table column config to use mantine

### PR DESCRIPTION
**Pull request by magnew**
Closes: #6255 

### Description:

Migrates the table column/pivot config to Mantine. 

### Acceptance criteria:
- [ ] update drag and drop to use mantine components and mantine icons
- [ ] update field inputs with hide/lock functionality
- [ ] sections that should be included to the migration:
  - [ ] columns
  - [ ] rows
  - [ ] metrics

### CODE REVIEW

**Review by yasmin-e — COMMENTED** [review]
Left some comments, nothing major 👍🏼  

I noticed this weird bug with the checkboxes while testing, I don't expect it to be fixed in this PR just thought I'd let you know since you are working on migrating config in general
The checkboxes don't have a pointer cursor on hover and when you click they flicker 
Spooky! 👻 
https://github.com/lightdash/lightdash/assets/67699259/4370cc85-b232-4917-a7c7-c157e009c2eb

**Comment by joaoviana** [comment]
https://github.com/lightdash/lightdash/assets/7611706/e4ea5f9a-89cc-42c6-826e-5f419baa50c9

Nice stuff, have a few Qs here that I'm not sure if introduced on this PR or not...

while dragging items, would there be a way to have a way to still see the container elements so users don't lose context? when it's grayed out I have no idea of what was there before 😆 

Now regarding the popover when clicking `Configure`, could we add an offset so there's some margin to the right? 

Also, the tabs' underline seems a bit long.

What do you think? Don't want to increase scope in this PR so feel free to manage it as you see fit 👍

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/ColumnConfiguration.tsx [hunk 1]
```diff
@@ -23,90 +23,107 @@ const ColumnConfiguration: React.FC<{ fieldId: string }> = ({ fieldId }) => {
...
-                        position="top-right"
-                        content={
-                            isPivotingDimension
-                                ? "Can't hide pivot dimensions"
-                                : disableHidingDimensions
-                                ? 'Cannot hide dimensions when pivoting'
-                                : isColumnVisible(fieldId)
-                                ? 'Hide column'
-                                : 'Show column'
+        <Group spacing="xs" noWrap style={{ width: '100%' }}>
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/ColumnConfiguration.tsx:34]: Weirdly it doesn't work here. I don't know why, but I tried it a few different ways to get it to work and couldn't. Setting it with the `sx` prop doesn't work either. There must be a bug with their internal emotion?

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/ColumnConfiguration.tsx [hunk 2]
```diff
@@ -23,90 +23,107 @@ const ColumnConfiguration: React.FC<{ fieldId: string }> = ({ fieldId }) => {
...
-                                ) : (
-                                    <IconEye size={18} />
-                                )
+                        variant="light"
+                        onClick={() => {
+                            // Close the tooltip. This click causes a lot of
+                            // re-rendering and it's easy to move the mouse away
+                            // before getting mouse events back, so the tooltip
+                            // stays open. Closing it is a workaround, but render
+                            // perf could use some work.
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/ColumnConfiguration.tsx:75]: Sure!

**Comment by IrakliJani** [comment]
@joaoviana nice feedback on UI and UX. let's not introduce UI changes since this is a purely a refactor to move asap from blueprint. as of UX

agreed on the items hiding when you start dragging

**Comment by magnew** [comment]
> review-drag-drop-and-tabs.mov 
> Nice stuff, have a few Qs here that I'm not sure if introduced on this PR or not...
> 
> while dragging items, would there be a way to have a way to still see the container elements so users don't lose context? when it's grayed out I have no idea of what was there before 😆
> 
> Now regarding the popover when clicking `Configure`, could we add an offset so there's some margin to the right?
> 
> Also, the tabs' underline seems a bit long.
> 
> What do you think? Don't want to increase scope in this PR so feel free to manage it as you see fit 👍

I think this is all good feedback, but I don't think most of it is introduced in this PR. It's basically a pure port of what is there, and I think we should keep it that way for now. That said, I agree, and lets make these changes. To address them individually:

- Yea, the drag hiding behavior is a bit strange. I'll file a ticket to see if we can keep the items around for context. My guess is that it's this way to make it very clear that there are only two areas to drop it. I bet if you keep the other items there, it will seem like you can reorder them with drag (partially cause you can elsewhere). Still, I agree it's weird -- I'll file a ticket. 
- I totally agree with the fact that the popup feels super strange smashed against the side! Feels very odd. But this is existing behavior. I'm not going to file a ticket cause I know Priyanka is working on new designs for this anyway. It's going to be a drawer. 
- Tabs underline - I respectfully disagree 🙂. The typical pattern with horizontal tabs is to divide the space where the tabs are from the space below. If the underline doesn't go across the panel it feels strange. I'm willing to be convinced to change it if you have strong feelings.

**Comment by magnew** [comment]
> Left some comments, nothing major 👍🏼
> 
> I noticed this weird bug with the checkboxes while testing, I don't expect it to be fixed in this PR just thought I'd let you know since you are working on migrating config in general The checkboxes don't have a pointer cursor on hover and when you click they flicker Spooky! 👻
> 
>  Screen.Recording.2023-07-20.at.11.04.49.mov

Woah, weird. Good catch! the checkboxes are not part of this review, but I didn't notice those issue. I will look into it and open another PR. Thanks!

**Review by IrakliJani — APPROVED** [review]
code looks good, tested functionality and could not find any issues! 🚢

**Pull request merged**